src/main/java/github/scarsz/discordsrv/hooks/permissions/LuckPermsHook.java

```diff
@@ -149,6 +150,12 @@ public void calculate(@NonNull Player target, net.luckperms.api.context.ContextC
             consumer.accept(CONTEXT_ROLE, role.getName());
             consumer.accept(CONTEXT_ROLE_ID, role.getId());
         }
+
+        for (Guild guild : member.getJDA().getGuilds()) {
+            if (guild.getMember(member.getUser()) == null) continue;
+
+            consumer.accept(CONTEXT_SERVER_ID, guild.getId());
```

Comment:
This would not run if the user is not in the 'main guild', seems kind of counter intuitive